src/HttpKernel/Controller/RectorRunValueResolver.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Website\HttpKernel\Controller;
+
+use Iterator;
+use Ramsey\Uuid\Uuid;
+use Rector\Website\Entity\RectorRun;
+use Rector\Website\Repository\RectorRunRepository;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+
+final class RectorRunValueResolver implements ArgumentValueResolverInterface
+{
+    /**
+     * @var RectorRunRepository
+     */
+    private $rectorRunRepository;
+
+    public function __construct(RectorRunRepository $rectorRunRepository)
+    {
+        $this->rectorRunRepository = $rectorRunRepository;
+    }
+
+    public function supports(Request $request, ArgumentMetadata $argumentMetadata): bool
+    {
+        if ($argumentMetadata->getType() !== RectorRun::class) {
+            return false;
+        }
+
+        $argumentValue = $request->get($argumentMetadata->getName());
+
+        return $argumentValue !== null;
```

Comment:
Can we add here that `Uuid::isValid($argumentValue)` ? It would prevent InvalidUuidException

Reply:
Not sure what exactly you mean, but sure. Send PR please.

In the end all those cases should end in 404 I guess